src/EEG_1SCU_CM.py

```diff
@@ -1,177 +1,182 @@
+            batch_idx (int): Index of the batch.
+
+        Returns:
+            torch.Tensor: Training loss tensor.
+        """
+        x, y = batch
+        y_hat = self.model(x)
+        loss = self.criterion(y_hat, y.long())
+        self.log("train_loss", loss, prog_bar=True)
+        return loss
+
+    def test_step(self, batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int):
+        """
+        Defines the test step logic.
+
+        Args:
+            batch (Tuple[torch.Tensor, torch.Tensor]): A tuple containing input data and corresponding labels.
+            batch_idx (int): Index of the batch.
+        """
+        x, y = batch
+        y_hat = self.model(x)
+        test_loss = self.criterion(y_hat, y.long())
+        test_acc = self.test_acc(torch.argmax(y_hat, dim=1), y.long())
+        self.test_step_outputs.append(
+            {"test_loss": test_loss, "test_acc": test_acc, "y": y, "y_hat": y_hat}
+        )
+
+    def on_test_epoch_end(self) -> None:
+        """
+        Calculates and logs test loss and accuracy at the end of each testing epoch.
+        """
+        test_loss = torch.stack([x["test_loss"] for x in self.test_step_outputs]).mean()
+        self.log("test_loss", test_loss, on_epoch=True)
+
+        mean_acc = torch.tensor([x["test_acc"] for x in self.test_step_outputs]).mean()
+        self.log("test_acc", mean_acc, on_epoch=True)
+
+        #  Extract true labels and predicted labels for CCM function
+        cnf_labels = np.concatenate(
+            [x["y"].cpu().numpy() for x in self.test_step_outputs]
+        )
+        cnf_raw_scores = np.concatenate(
+            [x["y_hat"].cpu().numpy() for x in self.test_step_outputs]
+        )
+
+        # Apply softmax to raw scores to obtain probabilities
+        cnf_probs = torch.softmax(torch.tensor(cnf_raw_scores), dim=1)
+
+        # Get predicted labels by selecting the class with the highest probability
+        cnf_predictions = np.argmax(cnf_probs, axis=1)
+
+        # Call your CCM function to plot the confusion matrix
+        CCM(cnf_labels, cnf_predictions)
+
+    def configure_optimizers(self) -> optim.Optimizer:
+        """
+        Configures the optimizer used for training the model.
+
+        Returns:
+            torch.optim.Optimizer: The optimizer instance.
+        """
+        optimizer = optim.Adam(self.parameters(), lr=self.config["learning_rate"])
+        return optimizer
+
+
+def main(args):
+    config_file = args.config_file
+    config = yaml.safe_load(open(config_file))
+
+    input_data = load_data()
+    input_label = load_label()
+
+    datamodule = SCU_DataModule(
+        input_data, input_label, config["batch_size"], args.seed_n
+    )
+    model = SCU_Model(config)
+
+    trainer = pl.Trainer(
+        max_epochs=config["num_epochs"],
+        logger=True,
+        log_every_n_steps=1,
+        accelerator="cpu",
+    )
```

Comment:
you could allow some more options to be set here including setting mixed precision etc

Reply:
Will add issue for that